src/openrct2/EditorObjectSelectionSession.cpp

```diff
@@ -192,14 +192,10 @@ void setup_in_use_selection_flags()
         }
     } while (tile_element_iterator_next(&iter));
 
-    for (ride_id_t ride_index = 0; ride_index < MAX_RIDES; ride_index++)
+    for (auto& ride : GetRideManager())
     {
-        auto ride = get_ride(ride_index);
-        if (ride != nullptr)
-        {
-            ObjectEntryIndex type = ride->subtype;
-            Editor::SetSelectedObject(ObjectType::Ride, type, OBJECT_SELECTION_FLAG_SELECTED);
-        }
+        ObjectEntryIndex type = ride.subtype;
+        Editor::SetSelectedObject(ObjectType::Ride, type, OBJECT_SELECTION_FLAG_SELECTED);
```

Comment:
We should make a refactor item for iterating over all rides.

We should also make one for iterating over a tile.